kore/src/Kore/Step/Simplification/ExpandAlias.hs

```diff
@@ -34,25 +27,21 @@ import Kore.Internal.TermLike (
 import Kore.Rewriting.RewritingVariable (
     RewritingVariableName,
  )
-import Kore.Unification.Unify (
-    MonadUnify,
- )
 import Prelude.Kore
 
-expandAlias ::
-    forall unifier.
-    MonadUnify unifier =>
-    ( TermLike RewritingVariableName ->
-      TermLike RewritingVariableName ->
-      MaybeT unifier (Pattern RewritingVariableName)
-    ) ->
+data UnifyExpandAlias = UnifyExpandAlias
+    { term1, term2 :: !(TermLike RewritingVariableName)
+    }
+
+matchExpandAlias ::
     TermLike RewritingVariableName ->
     TermLike RewritingVariableName ->
-    MaybeT unifier (Pattern RewritingVariableName)
-expandAlias recurse t1 t2 =
+    Maybe UnifyExpandAlias
+matchExpandAlias t1 t2 =
```

Comment:
What is the corresponding function for this one, that carries out unification?

Reply:
The terms are fed back into `maybeTermAnd` [here](https://github.com/kframework/kore/blob/2a9962c6b9b742952e4ccdb134ae42b823ff8b4f/kore/src/Kore/Step/Simplification/AndTerms.hs#L238).